cpp/tests/hashing/bloom_filter_test.cu

```diff
@@ -0,0 +1,91 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// Remove this after cuco bump
+#include "src/io/parquet/arrow_filter_policy.cuh"
+
+#include <cudf_test/base_fixture.hpp>
+#include <cudf_test/column_utilities.hpp>
+#include <cudf_test/column_wrapper.hpp>
+#include <cudf_test/iterator_utilities.hpp>
+
+#include <cudf/detail/cuco_helpers.hpp>
+#include <cudf/hashing.hpp>
+#include <cudf/hashing/detail/xxhash_64.cuh>
+#include <cudf/utilities/default_stream.hpp>
+
+#include <cuco/bloom_filter.cuh>
+
+using StringType = cudf::string_view;
+
+class BloomFilterTest : public cudf::test::BaseFixture {};
+
+TEST_F(BloomFilterTest, TestStrings)
+{
+  using key_type    = StringType;
+  using hasher_type = cudf::hashing::detail::XXHash_64<key_type>;
+  using policy_type = cuco::arrow_filter_policy<key_type, hasher_type>;
+  using word_type   = policy_type::word_type;
+
+  std::size_t constexpr num_filter_blocks = 4;
+  auto stream                             = cudf::get_default_stream();
+
+  // strings data
+  auto data = cudf::test::strings_column_wrapper(
+    {"seventh",    "fifteenth",   "second",      "tenth",      "fifth",       "first",
+     "seventh",    "tenth",       "ninth",       "ninth",      "seventeenth", "eighteenth",
+     "thirteenth", "fifth",       "fourth",      "twelfth",    "second",      "second",
+     "fourth",     "seventh",     "seventh",     "tenth",      "thirteenth",  "seventeenth",
+     "fifth",      "seventeenth", "eighth",      "fourth",     "second",      "eighteenth",
+     "fifteenth",  "second",      "seventeenth", "thirteenth", "eighteenth",  "fifth",
+     "seventh",    "tenth",       "fourteenth",  "first",      "fifth",       "fifth",
+     "tenth",      "thirteenth",  "fourteenth",  "third",      "third",       "sixth",
+     "first",      "third"});
+  auto d_column = cudf::column_device_view::create(data);
+
+  // Spawn a bloom filter
+  cuco::bloom_filter<key_type,
+                     cuco::extent<size_t>,
+                     cuda::thread_scope_device,
+                     policy_type,
+                     cudf::detail::cuco_allocator<char>>
+    filter{num_filter_blocks,
+           cuco::thread_scope_device,
+           {hasher_type{cudf::DEFAULT_HASH_SEED}},
+           cudf::detail::cuco_allocator<char>{rmm::mr::polymorphic_allocator<char>{}, stream},
+           stream};
+
+  // Add strings to the bloom filter
+  filter.add(d_column->begin<key_type>(), d_column->end<key_type>(), stream);
+
+  // Number of words in the filter
+  cudf::size_type const num_words = filter.block_extent() * filter.words_per_block;
+
+  auto const output = cudf::column_view{
+    cudf::data_type{cudf::type_id::UINT32}, num_words, filter.data(), nullptr, 0, 0, {}};
+
+  // Expected filter bitset words computed using Arrow implementation here:
+  // https://godbolt.org/z/oKfqcPWbY
```

Comment:
Since this is a pure C++ reproducer with no dependencies, should we reuse that implementation here rather than comparing to hardcoded values? Seems like that would make it easier to include tests for other types, too.

(Trying to prevent us from depending on godbolt.org being online to be able to recreate these values)

Reply:
We are testing other types in cuco and only needed to test for `string_view` (string-like) type here as it is not available in cuco. In ideal case, this test should be moved to cuco as well.

I opted for this solution (similar to the one in cuco) as the godbolt reproducer uses select modified files from arrow cpp repo with a main function to get these values. Doing the same in libcudf might not be ideal. 

In light of this, maybe we can either:

- Keep this as is and create an issue to move this to cuco once we have `string_view` or (span-like) support there.
- Remove this test from cudf altogether and add it back in cuco later - I would like keep the test active somewhere to ensure the bloom filter works correctly for `string_view` type. i.e. hash the entire string and insert the final hash instead of hashing each byte and inserting it.

Reply:
how does cuco test the bloom filters?

Reply:
Same as here. We insert a bunch of int64, int32, double keys etc to bloom filters using the same `godbolt` setup and compare the bitset words by inserting the same values to corresponding cuco::bloom_filters.